### PR TITLE
feat(PRU-ICSS): Add documentation for PTP in switch mode

### PR DESCRIPTION
**Pull request by danish-ti**
Add documentation for running PTP to synchronise time between master and slave DUTs in switch mode.

### CODE REVIEW

**Review by StaticRocket — CHANGES_REQUESTED** [review]
Over 50% of this file was using the proper 3 space indentation prior to this PR. Please standardize.

In addition, the rstcheck warning is valid. Lists should be indented to more appropriately indicate where they begin an end.

File: source/linux/Foundational_Components/PRU-ICSS/Linux_Drivers/PRU_ICSSG_Ethernet_Switch.rst [hunk 1]
```diff
@@ -166,6 +166,196 @@ To disable cut through on all queues
 
    devlink dev param set platform/icssg1-eth name cut_thru value 0 cmode runtime
 
+PTP
+"""
+PTP can be run during switch mode. To run PTP in switch mode, 3 AM64x EVMs need to be connected with middle EVM acting as the switch. Use the following ptp config file on all slave EVMs. For Master EVM, priority1 filed needs to be changed.
+
```
Comment (nsaulnier-ti) [inline source/linux/Foundational_Components/PRU-ICSS/Linux_Drivers/PRU_ICSSG_Ethernet_Switch.rst:172]: Should we also point customers to the PRU Ethernet and CPSW PTP documentation?
https://software-dl.ti.com/processor-sdk-linux/esd/AM64X/10_00_07_04/exports/docs/linux/Foundational_Components/PRU-ICSS/Linux_Drivers/PRU_ICSSG_Ethernet.html#ptp-ordinary-clock
https://software-dl.ti.com/processor-sdk-linux/esd/AM64X/10_00_07_04/exports/docs/linux/Foundational_Components/Kernel/Kernel_Drivers/Network/CPSW-PTP.html

The CPSW page in particular is a pretty long and detailed document. I haven't tried it out myself, but I assume most of the concepts also apply to PRU Ethernet?

Comment (danish-ti) [reply source/linux/Foundational_Components/PRU-ICSS/Linux_Drivers/PRU_ICSSG_Ethernet_Switch.rst:172]: @nsaulnier-ti The PRU-ICSS documentation only documents the PTP feature in MAC mode. I am adding details on PTP feature in switch mode in PRU_ICSSG_Ethernet_Switch.rst.

The CPSW PTP documentation is very detailed however not all of it is applicable as it is to PRU-ICSS Switch. Eg - to make sure ptp packets are forwarded by switch, the command `bridge mdb add dev br0 port br0 grp $ptp_dst_mac permanent vid 1` is needed. This is not the case for CPSW. I am planning to have a separate section for PTP for PRU-ICSS. Currently I have only added one PTP case. I will add more scenarios for running in PTP in switch mode as we test and verify. So I would like to keep the section separate if it's OK.

File: source/linux/Foundational_Components/PRU-ICSS/Linux_Drivers/PRU_ICSSG_Ethernet_Switch.rst [hunk 2]
```diff
@@ -166,6 +166,196 @@ To disable cut through on all queues
...
+**gPTP.cfg**
+
+.. code-block:: console
+
```
[the remark this answers is not in the document]
Comment (danish-ti) [reply source/linux/Foundational_Components/PRU-ICSS/Linux_Drivers/PRU_ICSSG_Ethernet_Switch.rst:176]: @StaticRocket  Sure will update.

**Review by StaticRocket — APPROVED** [review]

**Review by cshilwant — APPROVED** [review]

File: source/linux/Foundational_Components/PRU-ICSS/Linux_Drivers/PRU_ICSSG_Ethernet_Switch.rst (hunk 1 above)
Comment (nsaulnier-ti) [reply source/linux/Foundational_Components/PRU-ICSS/Linux_Drivers/PRU_ICSSG_Ethernet_Switch.rst:172]: Good to hear that you are planning on adding additional PTP documentation in the future. In that case the current documentation looks fine for now.

**Pull request merged**